certify

```diff
@@ -62,10 +68,182 @@ else
 	skipRegExp="Pre-provisioned|Inline|Disruptive|$skipRegExp"	
 fi
 
+# Find bash version
+bashVer="$(echo $BASH_VERSION)"
+case "${bashVer}" in
+    3.2*)     	bashVersion=3;;
+    4*)    		bashVersion=4;;
+    *)          bashVersion="UNKNOWN:$BASH_VERSION"
+esac
+
 
 runTests()
 {
-	go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=$kubeconfig --driverdef=$driverDefPath -ginkgo.skip=$skipRegExp -timeout=0
+	OUTPUT=$(go test -v ./cmd/... -ginkgo.v -ginkgo.progress --kubeconfig=$kubeconfig --driverdef=../../$driverDefPath -ginkgo.skip=$skipRegExp -timeout=0 | tee /dev/tty)	
+	
+	printf "\n\n\n\n"
+        center "CERTIFICATION SUMMARY"
+
+	printf "\n\n${PURPLE}Kubernetes Version :${NC} $(kubectl version --kubeconfig=$kubeconfig | grep "Server Version" | grep -o 'GitVersion:".*"' | cut -d \" -f2)\n"
+
+	printf "\n\n${PURPLE}Driver Info${NC}\n"
+	printHorizontalDash 15
+	printf "\n"
+	
+	cat $driverDefPath	
+	generateMatrix "$OUTPUT"
+}
+
+generateMatrix()
+{
+	printf "\n\n"
+	rawOutput=$1
+
+	passedTestPatterns=$(echo "$rawOutput" | awk 'c&&!--c;/SLOW TEST/{c=3}')
+	passedTestDescriptions=$(echo "$rawOutput" | awk 'c&&!--c;/SLOW TEST/{c=5}')
+	failedTestPatterns=$(echo "$rawOutput" | awk '/Failures/{y=1;next}y' | grep -G "\[Testpattern:[^\[]*" -o)
+	failedTestDescriptions=$(echo "$rawOutput" | awk '/Failures/{y=1;next}y' | grep -G "\[It\].*" -o)
+
+	# converting string to an array is different from bash 3.2 to bash 4
+	case "$bashVersion" in
+		"4")		readarray passedTestPatternsArray < <(printf "$passedTestPatterns")
+					readarray passedTestDescriptionsArray < <(printf "$passedTestDescriptions")
+					readarray failedTestPatternArray < <(printf "$failedTestPatterns")
+					readarray failedTestDescriptionsArray < <(printf "$failedTestDescriptions")
+					;;
+		"3")		IFS=$'\n' read -rd '' -a passedTestPatternsArray <<<"$passedTestPatterns"
+					IFS=$'\n' read -rd '' -a passedTestDescriptionsArray <<<"$passedTestDescriptions"
+					IFS=$'\n' read -rd '' -a failedTestPatternArray <<<"$failedTestPatterns"
+					IFS=$'\n' read -rd '' -a failedTestDescriptionsArray <<<"$failedTestDescriptions"
+					;;
+		*)			printf "Error: Unsupport bash version\n"
```

Comment:
can we default to version 3 instead of throwing an error? I'm not familiar enough with the differences between versions, but maybe the method used in version 3 bash also probably works for lower versions. Also, we probably won't run into much versions lower than 3 I'm guessing, so might be safe to assume if not version 4 just use what you have in version 3 for all other versions??

Reply:
ya that makes sense, i'll fix that

Reply:
done